Google Books.js

```diff
@@ -116,17 +116,22 @@ function parseXML(text) {
 	}
 	
 	var ISBN;
-	const ISBN10Re = /(ISBN:)(\w{10})$/;
-	const ISBN13Re = /(ISBN:)(\w{13})$/;
+	const ISBN10Re = /(?:ISBN:)(\w{10})$/;
+	const ISBN13Re = /(?:ISBN:)(\w{13})$/;
+	const booksIDRe = /^(\w{12})$/;
 	var identifiers = ZU.xpath(xml, "dc:identifier", ns);
 	for (var i in identifiers) {
 		var ISBN10Match = ISBN10Re.exec(identifiers[i].textContent);
 		var ISBN13Match = ISBN13Re.exec(identifiers[i].textContent);
+		var booksIDMatch = booksIDRe.exec(identifiers[i].textContent);
 		if (ISBN10Match != null) {
-			ISBN = ISBN10Match[2];
+			ISBN = ISBN10Match[1];
 		}
 		if (ISBN13Match != null) {
-			ISBN = ISBN13Match[2];
+			ISBN = ISBN13Match[1];
+		}
+		if (booksIDMatch != null) {
+			newItem.url = 'https://books.google.com/books?id=' + booksIDMatch[1];
```

Comment:
Correct me if I'm wrong, but there are two use cases here. (1) Google Books is used to lookup ISBNs for metadata, no? And in those cases you don't want to pull the URL as there isn't necessarily any text cited at that time. (2) Google Books often _does_ have full-page view (if not full text) for recent and public domain books, and in those cases it would absolutely be useful to have the URL linked in the citation.

Case #2 is much more frequent for me, but then again I'm less privy to the backend on ISBN lookup. Wouldn't it make sense to handle the two cases differently, though, based on whether the user is citing from the page view (more than the snippet view)? It would even be nice to save the full-page view image to Z for citation...